Expensify/App
# Refactor picker to be a cross-platform component

@nickmurray47 @NikkiWines This PR was based on some stuff that I came across in https://github.com/Expensify/Expensify.cash/pull/1767, primarily some missing method docs. As I looked more at the PR, I spotted a few things that should be improved to match our existing patterns better. Please have a look at this and let me know if you have any questions

### Details
- Adds some missing method docs
- Renames a prop to have a less redundant name
- DRY up some logic in the checkbox
- Convert the picker into a cross-platform component

### Fixed Issues
No issue, just some things I spotted in https://github.com/Expensify/Expensify.cash/pull/1767

### Tests / QA Steps
1. Go to the profile page
2. Go to preferences
3. Change the priority mode
4. Go back to settings
5. Go to profile
6. Change preferred prounouns
7. Change timezones
8. Change automatic timezone setting
9. Verify all changes are saved  properly

### Tested On


### Screenshots


#### Web
![image](https://user-images.githubusercontent.com/1228807/113749844-572a9980-96c7-11eb-9168-fda519cd1518.png)


#### Mobile Web
![image](https://user-images.githubusercontent.com/1228807/114237236-040f4b80-9940-11eb-894e-92342ff5e898.png)


#### Desktop
![image](https://user-images.githubusercontent.com/1228807/113749875-5f82d480-96c7-11eb-859a-8437c61af57d.png)


#### iOS
![image](https://user-images.githubusercontent.com/1228807/113753107-f3a26b00-96ca-11eb-8ea5-1a3b95d1d4be.png)


#### Android
![image](https://user-images.githubusercontent.com/1228807/113749860-5bef4d80-96c7-11eb-9169-aaf98a2bd148.png)
